tests/unit/customizations/history/test_db.py

```diff
@@ -598,6 +538,64 @@ def test_can_serialize_unknown_type(self):
         reloaded = json.loads(string_value)
         self.assertEqual(encoded, reloaded)
 
+    def test_can_serialize_non_utf_8_bytes_type(self):
+        original = b'\xfe\xed'  # Non utf-8 byte squence
+        encoded = '<Byte sequence>'
+        string_value = json.dumps(original, cls=PayloadSerializer)
+        reloaded = json.loads(string_value)
+        self.assertEqual(encoded, reloaded)
+
+    def test_does_preserve_utf_8_bytes_type(self):
+        original = b'foobar'  # utf-8 byte squence
+        encoded = 'foobar'
+        string_value = json.dumps(original, cls=PayloadSerializer)
+        reloaded = json.loads(string_value)
+        self.assertEqual(encoded, reloaded)
+
+    def test_can_serialize_non_utf_8_bytes_type_in_dict(self):
+        original = {'foo': 'bar', 'bytes': b'\xfe\xed'}
+        encoded = {'foo': 'bar', 'bytes': '<Byte sequence>'}
+        string_value = json.dumps(original, cls=PayloadSerializer)
+        reloaded = json.loads(string_value)
+        self.assertEqual(encoded, reloaded)
+
+    def test_can_serialize_non_utf_8_bytes_type_in_list(self):
+        original = ['foo', b'\xfe\xed']
+        encoded = ['foo', '<Byte sequence>']
+        string_value = json.dumps(original, cls=PayloadSerializer)
+        reloaded = json.loads(string_value)
+        self.assertEqual(encoded, reloaded)
+
+    def test_can_serialize_non_utf_8_bytes_type_in_tuple(self):
+        original = ('foo', b'\xfe\xed')
+        encoded = ['foo', '<Byte sequence>']
```

Comment:
Should this remain a tuple?

Reply:
No its being serialzied to JSON which turns it into a list.